File: 7-kyu/two-fighters-one-winner/two-fighters-one-winner.py
class Fighter(object):
    def __init__(self, name, health, damage_per_attack):
        self.name = name
        self.health = health
        self.damage_per_attack = damage_per_attack
        
    def __str__(self): return "Fighter({}, {}, {})".format(self.name, self.health, self.damage_per_attack)
    __repr__=__str__
​
​
def declare_winner(fighter1, fighter2, first_attacker):
    the_attacker = fighter1 if fighter1.name == first_attacker else fighter2
    the_defender = fighter2 if fighter1.name == first_attacker else fighter1
    
    while fighter1.health > 0 and fighter2.health > 0:
        the_defender.health -= the_attacker.damage_per_attack
        
        if the_defender.health <= 0:
            return the_attacker.name
    
        the_defender, the_attacker = the_attacker, the_defender
        

File: 7-kyu/two-fighters-one-winner/two-fighters-one-winner.js
   function Fighter(name, health, damagePerAttack){
    this.name = name;
    this.health = health;
    this.damagePerAttack = damagePerAttack;
    this.toString = function() { return this.name; }
  };
​
function declareWinner(fighter1, fighter2, firstAttacker) {
      let off = fighter1.name === firstAttacker ? fighter1 : fighter2;
      let def = fighter1.name === firstAttacker ? fighter2: fighter1;
  
    while(fighter1.health > 0 && fighter2.health > 0){
      def.health -= off.damagePerAttack;
      
      if( def.health <= 0 ){
        return off.name;
      }
    [off, def] = [def, off]
    }
};